### ISSUE TITLE
Installed via docker-compose on unraid. 5X'ed my cpu util, EXTREMELY slow

### ISSUE THREAD

**Issue by superdong69** [manyfold3d/manyfold#2419]
**Describe the bug**
Was routinely using 18% of my CPU on an R710 server, now using 80% util. 
Collection is over 18TB. Created the library 5 days ago, took 5 days to finish. Is completely unusable.

Using postgres15 Server DB connection.

**To Reproduce**
Try to navigate to anything on the app. Just sits there thinking. Is there a way to fix this, or is this a limitation of the software?

**Comment by Floppy** [comment]
That's a big collection, so it's probably doing a lot of analysis work. Hard to tell though without knowing more... when you say it was using 18%, was that a previous version of Manyfold or is this the first time you've used it?

It's probably a good idea to set DEFAULT_WORKER_CONCURRENCY to 1, and you could also set PERFORMANCE_WORKER_CONCURRENCY to 0 while we get it under control. Could even set the default worker concurrency to zero as well to be sure.

There *are* some performance issues with large libraries, and this is the largest I've heard of so far, so we might be hitting that. It could potentially be due to a huge number of autogenerated tags, perhaps.

Once you've changed the background worker concurrency, the idle cpu use should drop, and it should become semi-usable, though if the tag problem is the root cause then it might still be very slow to respond.

If you can, take a look at the /admin area and see how many model, files and tags it's indexed, that will give us some clues.

**Comment by superdong69** [comment]
So here's a bit of info.
I've never used this before. So my server without it running was running at a max of 18% CPU across all threads.
Where would I put these changes in you suggested? I'd really love to have this work normally, because I need it badly. LOL

Here's a brief example of tags, I couldn't find just how many tags I have, but on the library page, on the right, I have to scroll down for about 5 minutes to get to the bottom of the page, there are so many tags.
Here's a brief example.
[!new (81774)
[patreon (22796)] 
[2021 (22566)] 
[2022 (19270)] 
[2023 (13633)]  
[supported (9950)]
[2020 (9614)]
[kickstarter (6124)]
[unsupported (4402)]
[pre supported (3697]

**Comment by Floppy** [comment]
Ok, sounds like there are a lot, that's probably the issue. The concurrency stuff goes in environment variables, but I don't think that's your problem, so don't worry about it. Let me have a go at improving tag performance.

**Comment by Floppy** [comment]
Yep, pretty certain after some investigation that this is tag performance issues. I've got a handle on it and am making some improvements.

**Comment by superdong69** [comment]
Awesome! Can't wait!

**Comment by Floppy** [comment]
I've release v0.74.0 with a bunch of tag performance improvements; hopefully this will help! I'd also recommend changing your tag visibility threshold to at least 2 (the new default) in settings.

**Comment by superdong69** [comment]
Roger that! I'll try it and report back!

**Comment by superdong69** [comment]
Just updated.  Light years better.  Still overall load is still quite high.  Too high to leave running for sure.  Though it does occasionally drop to 30% it's more often just constantly cycling cpu into the 60% even though it's not actively doing anything.  Page response has improved significantly.  You're definitely on the way to something amazing. 

Please let me know if there is anything I can provide you to help improve.

**Comment by superdong69** [comment]
Correct.  At idle,  the overall cpu swings wildly, from 60% to 28%. Hovering most often north of 48%. Watching the individual cpus, they're getting thrashed.

**Comment by Floppy** [comment]
If CPU is high, it's likely that the background workers are active; the app consumes a lot of memory when quiescent (see #1784 for when I get round to fixing that), but it shouldn't use a lot of CPU, and mine doesn't here. I suspect that if you visit /sidekiq you'll see a bunch of work being done, probably by the performance worker?

**Comment by superdong69** [comment]
I'm not sure how to answer that. This is what I see in Sidekiq.

![image](https://github.com/user-attachments/assets/51025130-dea5-4913-a349-baaebd741f71)

Is there any other data I can get to you?

**Comment by superdong69** [comment]
As an aside, when I go into a model, and try to view the 3d rendering, it completely locks up my browser to crash.

![image](https://github.com/user-attachments/assets/dfdf5a9c-84d2-4c5a-a376-cf43d33e0079)

**Comment by superdong69** [comment]
And here you can see when I end the docker container:

![image](https://github.com/user-attachments/assets/ce48f38a-616c-402f-9dc0-5152214e4e83)

**Comment by Floppy** [comment]
Thanks! Doesn't look like background workers then, there's nothing running. Hmm. I will keep poking at it.
